src/daft-logical-plan/src/optimization/rules/push_down_projection.rs

```diff
@@ -148,12 +148,12 @@ impl PushDownProjection {
                     SourceInfo::Physical(external_info) => {
                         if required_columns.len() < upstream_schema.names().len() {
                             let pruned_upstream_schema = upstream_schema
-                                .fields
+                                .fields()
                                 .iter()
-                                .filter(|&(name, _)| required_columns.contains(name))
-                                .map(|(_, field)| field.clone())
+                                .filter(|field| required_columns.contains(&field.name))
+                                .cloned()
                                 .collect::<Vec<_>>();
-                            let schema = Schema::new(pruned_upstream_schema)?;
+                            let schema = Schema::new(pruned_upstream_schema);
```

Comment:
Depending on how often need to filter schema fields, it may be worth defining a filter method that accepts a lambda/predicate